internal/server/handlers/storage.go

```diff
@@ -139,41 +250,77 @@ func (c *StorageController) getValueHandler(e echo.Context) error {
 func (c *StorageController) getValueJSONHandler(e echo.Context) error {
 	body, err := io.ReadAll(e.Request().Body)
 	if err != nil {
-		log.Infof("something went wrong: %s", err.Error())
+		c.l.Infof("something went wrong: %s", err.Error())
+
 		return e.NoContent(http.StatusInternalServerError)
 	}
 
 	var mtrc metric.Metrics
+
 	err = json.Unmarshal(body, &mtrc)
 	if err != nil {
-		log.Infof("something went wrong: %s", err.Error())
+		c.l.Infof("something went wrong: %s", err.Error())
+
 		return e.NoContent(http.StatusInternalServerError)
 	}
 
-	m, ok := c.storage.GetValue(mtrc.MType, mtrc.ID)
-	if !ok {
+	v, err := c.storage.GetValue(e.Request().Context(), mtrc.MType, mtrc.ID)
+	if errors.Is(err, metric.ErrWrongMetricType) {
 		return e.NoContent(http.StatusNotFound)
 	}
 
-	return e.JSON(http.StatusOK, &m)
+	if err != nil {
+		c.l.Infof("something went wrong: %s", err.Error())
+
+		return e.NoContent(http.StatusInternalServerError)
+	}
+
+	if v.ID == "" {
+		return e.NoContent(http.StatusNotFound)
+	}
+
+	return e.JSON(http.StatusOK, v)
 }
 
 func (c *StorageController) getAllHandler(e echo.Context) error {
-	values := c.storage.GetAll()
+	values, err := c.storage.GetAll(e.Request().Context())
+	if err != nil {
+		c.l.Infof("something went wrong: %s", err.Error())
+
+		return e.NoContent(http.StatusInternalServerError)
+	}
 
 	builder := strings.Builder{}
 
 	builder.WriteString("<table><tr><th>Metric name</th><th>Metric type</th><th>Value</th></tr>")
+
 	for _, v := range values {
 		if v.Delta != nil {
-			builder.WriteString(fmt.Sprintf("<tr><td>%s</td><td>%s</td><td>%d</td></tr>", v.ID, v.MType, *v.Delta))
+			builder.WriteString(
```

Comment:
You can simplify this code. Please create a new variable for `Delta` and  `builder.WriteString` will be without `if`